{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9d2d3395-768c-4c4a-b416-15e6c67014dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Imports -------------------------\n",
    "import pandas as pd\n",
    "# ---------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9d22730c-3297-4bf5-9b3d-d06d9a3b9997",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Type</th>\n",
       "      <th>FGR</th>\n",
       "      <th>CD99</th>\n",
       "      <th>ARF5</th>\n",
       "      <th>SLC25A5</th>\n",
       "      <th>POLR2J</th>\n",
       "      <th>IL32</th>\n",
       "      <th>RPS20</th>\n",
       "      <th>IDS</th>\n",
       "      <th>...</th>\n",
       "      <th>MTATP6P1</th>\n",
       "      <th>LINC02972</th>\n",
       "      <th>CHCHD10</th>\n",
       "      <th>EEF1G</th>\n",
       "      <th>BRK1</th>\n",
       "      <th>C17orf49</th>\n",
       "      <th>LOC124903659</th>\n",
       "      <th>RPL17</th>\n",
       "      <th>NDUFA7</th>\n",
       "      <th>EGLN2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1058</td>\n",
       "      <td>1</td>\n",
       "      <td>5069.580040</td>\n",
       "      <td>1578.827545</td>\n",
       "      <td>2992.460441</td>\n",
       "      <td>3395.968681</td>\n",
       "      <td>1611.595663</td>\n",
       "      <td>2398.030516</td>\n",
       "      <td>18619.603402</td>\n",
       "      <td>3332.991606</td>\n",
       "      <td>...</td>\n",
       "      <td>19719.677898</td>\n",
       "      <td>4990.868852</td>\n",
       "      <td>373.719041</td>\n",
       "      <td>57813.834711</td>\n",
       "      <td>3010.063149</td>\n",
       "      <td>1122.335152</td>\n",
       "      <td>1186.151740</td>\n",
       "      <td>18162.339198</td>\n",
       "      <td>499.646110</td>\n",
       "      <td>3780.520158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1059</td>\n",
       "      <td>1</td>\n",
       "      <td>22322.926466</td>\n",
       "      <td>2977.917867</td>\n",
       "      <td>5103.243995</td>\n",
       "      <td>3036.767940</td>\n",
       "      <td>1406.053242</td>\n",
       "      <td>4941.322966</td>\n",
       "      <td>8062.225679</td>\n",
       "      <td>10545.204014</td>\n",
       "      <td>...</td>\n",
       "      <td>10154.439528</td>\n",
       "      <td>3324.101383</td>\n",
       "      <td>252.586598</td>\n",
       "      <td>26712.912781</td>\n",
       "      <td>2523.261991</td>\n",
       "      <td>1524.612704</td>\n",
       "      <td>220.297177</td>\n",
       "      <td>3295.036434</td>\n",
       "      <td>474.430542</td>\n",
       "      <td>6131.083961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1060</td>\n",
       "      <td>1</td>\n",
       "      <td>13309.274821</td>\n",
       "      <td>2625.227716</td>\n",
       "      <td>3452.118435</td>\n",
       "      <td>2699.675801</td>\n",
       "      <td>1713.541827</td>\n",
       "      <td>3821.111384</td>\n",
       "      <td>18238.310869</td>\n",
       "      <td>5798.275658</td>\n",
       "      <td>...</td>\n",
       "      <td>9669.544485</td>\n",
       "      <td>27923.568621</td>\n",
       "      <td>405.456436</td>\n",
       "      <td>47541.807734</td>\n",
       "      <td>3075.430963</td>\n",
       "      <td>1672.903361</td>\n",
       "      <td>743.801781</td>\n",
       "      <td>15782.782156</td>\n",
       "      <td>604.229611</td>\n",
       "      <td>5009.622187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1061</td>\n",
       "      <td>1</td>\n",
       "      <td>2092.509298</td>\n",
       "      <td>5217.403538</td>\n",
       "      <td>2348.651952</td>\n",
       "      <td>4755.579746</td>\n",
       "      <td>1902.448739</td>\n",
       "      <td>689.823337</td>\n",
       "      <td>32714.431318</td>\n",
       "      <td>2657.608782</td>\n",
       "      <td>...</td>\n",
       "      <td>18275.561505</td>\n",
       "      <td>2764.023179</td>\n",
       "      <td>307.642235</td>\n",
       "      <td>52272.074896</td>\n",
       "      <td>1992.220640</td>\n",
       "      <td>1298.331547</td>\n",
       "      <td>1388.079434</td>\n",
       "      <td>35091.990798</td>\n",
       "      <td>611.218713</td>\n",
       "      <td>2337.809935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1062</td>\n",
       "      <td>1</td>\n",
       "      <td>10993.829656</td>\n",
       "      <td>2478.139442</td>\n",
       "      <td>3871.937688</td>\n",
       "      <td>2526.369235</td>\n",
       "      <td>1892.057554</td>\n",
       "      <td>3776.799535</td>\n",
       "      <td>5219.417688</td>\n",
       "      <td>6519.792646</td>\n",
       "      <td>...</td>\n",
       "      <td>5777.510168</td>\n",
       "      <td>20434.521165</td>\n",
       "      <td>454.303864</td>\n",
       "      <td>13854.950602</td>\n",
       "      <td>3415.620723</td>\n",
       "      <td>1893.874800</td>\n",
       "      <td>680.118769</td>\n",
       "      <td>5727.192667</td>\n",
       "      <td>690.737793</td>\n",
       "      <td>5593.112143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>655</td>\n",
       "      <td>0</td>\n",
       "      <td>31504.607255</td>\n",
       "      <td>3255.911026</td>\n",
       "      <td>3730.017368</td>\n",
       "      <td>3048.706506</td>\n",
       "      <td>1048.029262</td>\n",
       "      <td>4071.288813</td>\n",
       "      <td>8457.360492</td>\n",
       "      <td>13688.423405</td>\n",
       "      <td>...</td>\n",
       "      <td>5938.515145</td>\n",
       "      <td>11401.686321</td>\n",
       "      <td>227.924972</td>\n",
       "      <td>25800.216417</td>\n",
       "      <td>2273.649598</td>\n",
       "      <td>1330.751430</td>\n",
       "      <td>434.642281</td>\n",
       "      <td>3956.475108</td>\n",
       "      <td>462.021279</td>\n",
       "      <td>5865.791959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>656</td>\n",
       "      <td>0</td>\n",
       "      <td>21039.133213</td>\n",
       "      <td>2778.658293</td>\n",
       "      <td>2275.656845</td>\n",
       "      <td>3030.174777</td>\n",
       "      <td>1112.282795</td>\n",
       "      <td>1083.601090</td>\n",
       "      <td>24764.319265</td>\n",
       "      <td>8841.901232</td>\n",
       "      <td>...</td>\n",
       "      <td>7708.204866</td>\n",
       "      <td>2321.749289</td>\n",
       "      <td>617.759784</td>\n",
       "      <td>66079.910808</td>\n",
       "      <td>3860.368282</td>\n",
       "      <td>1629.795294</td>\n",
       "      <td>339.893955</td>\n",
       "      <td>20161.586529</td>\n",
       "      <td>563.680588</td>\n",
       "      <td>5327.787309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>657</td>\n",
       "      <td>0</td>\n",
       "      <td>22588.818351</td>\n",
       "      <td>3438.052885</td>\n",
       "      <td>3856.288323</td>\n",
       "      <td>2660.885013</td>\n",
       "      <td>1174.076044</td>\n",
       "      <td>7760.914622</td>\n",
       "      <td>10425.208266</td>\n",
       "      <td>16723.906294</td>\n",
       "      <td>...</td>\n",
       "      <td>4138.766957</td>\n",
       "      <td>2581.266569</td>\n",
       "      <td>191.600944</td>\n",
       "      <td>26637.841436</td>\n",
       "      <td>2515.928494</td>\n",
       "      <td>1228.068759</td>\n",
       "      <td>572.657187</td>\n",
       "      <td>5504.128198</td>\n",
       "      <td>409.904026</td>\n",
       "      <td>7891.519010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>658</td>\n",
       "      <td>0</td>\n",
       "      <td>18752.923297</td>\n",
       "      <td>3383.866368</td>\n",
       "      <td>2979.236304</td>\n",
       "      <td>3475.303494</td>\n",
       "      <td>1069.758958</td>\n",
       "      <td>6539.832626</td>\n",
       "      <td>14655.016100</td>\n",
       "      <td>11304.538137</td>\n",
       "      <td>...</td>\n",
       "      <td>3496.915905</td>\n",
       "      <td>1633.697432</td>\n",
       "      <td>236.905281</td>\n",
       "      <td>38168.987424</td>\n",
       "      <td>2594.182098</td>\n",
       "      <td>1488.887348</td>\n",
       "      <td>459.956452</td>\n",
       "      <td>6856.094249</td>\n",
       "      <td>478.327002</td>\n",
       "      <td>5191.827723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>659</td>\n",
       "      <td>0</td>\n",
       "      <td>21653.214034</td>\n",
       "      <td>3285.588236</td>\n",
       "      <td>3140.755937</td>\n",
       "      <td>2982.100166</td>\n",
       "      <td>962.057182</td>\n",
       "      <td>7247.898317</td>\n",
       "      <td>9185.452866</td>\n",
       "      <td>9730.242227</td>\n",
       "      <td>...</td>\n",
       "      <td>4589.744961</td>\n",
       "      <td>39650.364469</td>\n",
       "      <td>338.675093</td>\n",
       "      <td>29649.402087</td>\n",
       "      <td>2561.112574</td>\n",
       "      <td>1495.228542</td>\n",
       "      <td>730.162136</td>\n",
       "      <td>11120.494057</td>\n",
       "      <td>382.049381</td>\n",
       "      <td>5795.120404</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows × 735 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ID  Type           FGR         CD99         ARF5      SLC25A5  \\\n",
       "0   1058     1   5069.580040  1578.827545  2992.460441  3395.968681   \n",
       "1   1059     1  22322.926466  2977.917867  5103.243995  3036.767940   \n",
       "2   1060     1  13309.274821  2625.227716  3452.118435  2699.675801   \n",
       "3   1061     1   2092.509298  5217.403538  2348.651952  4755.579746   \n",
       "4   1062     1  10993.829656  2478.139442  3871.937688  2526.369235   \n",
       "..   ...   ...           ...          ...          ...          ...   \n",
       "67   655     0  31504.607255  3255.911026  3730.017368  3048.706506   \n",
       "68   656     0  21039.133213  2778.658293  2275.656845  3030.174777   \n",
       "69   657     0  22588.818351  3438.052885  3856.288323  2660.885013   \n",
       "70   658     0  18752.923297  3383.866368  2979.236304  3475.303494   \n",
       "71   659     0  21653.214034  3285.588236  3140.755937  2982.100166   \n",
       "\n",
       "         POLR2J         IL32         RPS20           IDS  ...      MTATP6P1  \\\n",
       "0   1611.595663  2398.030516  18619.603402   3332.991606  ...  19719.677898   \n",
       "1   1406.053242  4941.322966   8062.225679  10545.204014  ...  10154.439528   \n",
       "2   1713.541827  3821.111384  18238.310869   5798.275658  ...   9669.544485   \n",
       "3   1902.448739   689.823337  32714.431318   2657.608782  ...  18275.561505   \n",
       "4   1892.057554  3776.799535   5219.417688   6519.792646  ...   5777.510168   \n",
       "..          ...          ...           ...           ...  ...           ...   \n",
       "67  1048.029262  4071.288813   8457.360492  13688.423405  ...   5938.515145   \n",
       "68  1112.282795  1083.601090  24764.319265   8841.901232  ...   7708.204866   \n",
       "69  1174.076044  7760.914622  10425.208266  16723.906294  ...   4138.766957   \n",
       "70  1069.758958  6539.832626  14655.016100  11304.538137  ...   3496.915905   \n",
       "71   962.057182  7247.898317   9185.452866   9730.242227  ...   4589.744961   \n",
       "\n",
       "       LINC02972     CHCHD10         EEF1G         BRK1     C17orf49  \\\n",
       "0    4990.868852  373.719041  57813.834711  3010.063149  1122.335152   \n",
       "1    3324.101383  252.586598  26712.912781  2523.261991  1524.612704   \n",
       "2   27923.568621  405.456436  47541.807734  3075.430963  1672.903361   \n",
       "3    2764.023179  307.642235  52272.074896  1992.220640  1298.331547   \n",
       "4   20434.521165  454.303864  13854.950602  3415.620723  1893.874800   \n",
       "..           ...         ...           ...          ...          ...   \n",
       "67  11401.686321  227.924972  25800.216417  2273.649598  1330.751430   \n",
       "68   2321.749289  617.759784  66079.910808  3860.368282  1629.795294   \n",
       "69   2581.266569  191.600944  26637.841436  2515.928494  1228.068759   \n",
       "70   1633.697432  236.905281  38168.987424  2594.182098  1488.887348   \n",
       "71  39650.364469  338.675093  29649.402087  2561.112574  1495.228542   \n",
       "\n",
       "    LOC124903659         RPL17      NDUFA7        EGLN2  \n",
       "0    1186.151740  18162.339198  499.646110  3780.520158  \n",
       "1     220.297177   3295.036434  474.430542  6131.083961  \n",
       "2     743.801781  15782.782156  604.229611  5009.622187  \n",
       "3    1388.079434  35091.990798  611.218713  2337.809935  \n",
       "4     680.118769   5727.192667  690.737793  5593.112143  \n",
       "..           ...           ...         ...          ...  \n",
       "67    434.642281   3956.475108  462.021279  5865.791959  \n",
       "68    339.893955  20161.586529  563.680588  5327.787309  \n",
       "69    572.657187   5504.128198  409.904026  7891.519010  \n",
       "70    459.956452   6856.094249  478.327002  5191.827723  \n",
       "71    730.162136  11120.494057  382.049381  5795.120404  \n",
       "\n",
       "[72 rows x 735 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NCounts_CIGT = pd.read_csv('Results/NCounts_CIGT.csv')\n",
    "NCounts_CIGT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "15e88937-e01f-4e61-a0ab-e586311168bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(964, 4)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mart_export = pd.read_csv('Datasets/mart_export.txt', sep='\\t')\n",
    "\n",
    "mart_export.rename(columns={\n",
    "    'Chromosome/scaffold name': 'Chromosome',\n",
    "    'Gene start (bp)': 'Start',\n",
    "    'Gene end (bp)': 'End',\n",
    "    'Gene name': 'Gene'\n",
    "}, inplace=True)\n",
    "\n",
    "mart_export = mart_export[['Chromosome', 'Start', 'End', 'Gene']]\n",
    "Genes_CIGT = NCounts_CIGT.columns.tolist()\n",
    "BED = mart_export[mart_export['Gene'].isin(Genes_CIGT)]\n",
    "BED.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e82f6bfc-3448-412f-8de7-fe31fb2ef0b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(720, 4)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def is_valid_chromosome(chromosome):\n",
    "    # Check if chromosome is \"X\" or \"Y\"\n",
    "    if chromosome in [\"X\", \"Y\"]:\n",
    "        return True\n",
    "    try:\n",
    "        # Attempt to convert chromosome to an integer\n",
    "        int(chromosome)\n",
    "        return True\n",
    "    except ValueError:\n",
    "        # If conversion fails, chromosome is not valid\n",
    "        return False\n",
    "\n",
    "BED = BED[BED['Chromosome'].apply(is_valid_chromosome)]\n",
    "BED['Chromosome'] = 'chr' + BED['Chromosome'].astype(str)\n",
    "BED.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7714ce65-8686-4a2d-af7c-29ad60698b65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Exp IDs</th>\n",
       "      <th>WGS Sample IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1059</td>\n",
       "      <td>BR2-106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1068</td>\n",
       "      <td>BR2-137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1073</td>\n",
       "      <td>BR2-158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1084</td>\n",
       "      <td>BR2-178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1085</td>\n",
       "      <td>BR2-201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>1079</td>\n",
       "      <td>BR2-1676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>1080</td>\n",
       "      <td>BR2-1732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>1081</td>\n",
       "      <td>BR2-1747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>1082</td>\n",
       "      <td>BR2-1783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>1083</td>\n",
       "      <td>BR2-1784</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Exp IDs WGS Sample IDs\n",
       "0           1059        BR2-106\n",
       "1           1068        BR2-137\n",
       "2           1073        BR2-158\n",
       "3           1084        BR2-178\n",
       "4           1085        BR2-201\n",
       "..           ...            ...\n",
       "91          1079       BR2-1676\n",
       "92          1080       BR2-1732\n",
       "93          1081       BR2-1747\n",
       "94          1082       BR2-1783\n",
       "95          1083       BR2-1784\n",
       "\n",
       "[96 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RNASeq_VCF_IDs = pd.read_excel('/home/wbd20/Ahmed/Datasets/RNA-Seq_VCF_IDs.xlsx')\n",
    "RNASeq_VCF_IDs['Gene Exp IDs'] = RNASeq_VCF_IDs['Gene Exp IDs'].astype('Int64')\n",
    "RNASeq_VCF_IDs.drop('RNA-seq Sample IDs', axis = 1, inplace = True)\n",
    "\n",
    "RNASeq_VCF_IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a5d8fd28-ccba-486c-a963-629218367b94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNASeq</th>\n",
       "      <th>WGS</th>\n",
       "      <th>Type</th>\n",
       "      <th>FGR</th>\n",
       "      <th>CD99</th>\n",
       "      <th>ARF5</th>\n",
       "      <th>SLC25A5</th>\n",
       "      <th>POLR2J</th>\n",
       "      <th>IL32</th>\n",
       "      <th>RPS20</th>\n",
       "      <th>...</th>\n",
       "      <th>MTATP6P1</th>\n",
       "      <th>LINC02972</th>\n",
       "      <th>CHCHD10</th>\n",
       "      <th>EEF1G</th>\n",
       "      <th>BRK1</th>\n",
       "      <th>C17orf49</th>\n",
       "      <th>LOC124903659</th>\n",
       "      <th>RPL17</th>\n",
       "      <th>NDUFA7</th>\n",
       "      <th>EGLN2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1058</td>\n",
       "      <td>1015</td>\n",
       "      <td>1</td>\n",
       "      <td>5069.580040</td>\n",
       "      <td>1578.827545</td>\n",
       "      <td>2992.460441</td>\n",
       "      <td>3395.968681</td>\n",
       "      <td>1611.595663</td>\n",
       "      <td>2398.030516</td>\n",
       "      <td>18619.603402</td>\n",
       "      <td>...</td>\n",
       "      <td>19719.677898</td>\n",
       "      <td>4990.868852</td>\n",
       "      <td>373.719041</td>\n",
       "      <td>57813.834711</td>\n",
       "      <td>3010.063149</td>\n",
       "      <td>1122.335152</td>\n",
       "      <td>1186.151740</td>\n",
       "      <td>18162.339198</td>\n",
       "      <td>499.646110</td>\n",
       "      <td>3780.520158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1059</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>22322.926466</td>\n",
       "      <td>2977.917867</td>\n",
       "      <td>5103.243995</td>\n",
       "      <td>3036.767940</td>\n",
       "      <td>1406.053242</td>\n",
       "      <td>4941.322966</td>\n",
       "      <td>8062.225679</td>\n",
       "      <td>...</td>\n",
       "      <td>10154.439528</td>\n",
       "      <td>3324.101383</td>\n",
       "      <td>252.586598</td>\n",
       "      <td>26712.912781</td>\n",
       "      <td>2523.261991</td>\n",
       "      <td>1524.612704</td>\n",
       "      <td>220.297177</td>\n",
       "      <td>3295.036434</td>\n",
       "      <td>474.430542</td>\n",
       "      <td>6131.083961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1060</td>\n",
       "      <td>1073</td>\n",
       "      <td>1</td>\n",
       "      <td>13309.274821</td>\n",
       "      <td>2625.227716</td>\n",
       "      <td>3452.118435</td>\n",
       "      <td>2699.675801</td>\n",
       "      <td>1713.541827</td>\n",
       "      <td>3821.111384</td>\n",
       "      <td>18238.310869</td>\n",
       "      <td>...</td>\n",
       "      <td>9669.544485</td>\n",
       "      <td>27923.568621</td>\n",
       "      <td>405.456436</td>\n",
       "      <td>47541.807734</td>\n",
       "      <td>3075.430963</td>\n",
       "      <td>1672.903361</td>\n",
       "      <td>743.801781</td>\n",
       "      <td>15782.782156</td>\n",
       "      <td>604.229611</td>\n",
       "      <td>5009.622187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1061</td>\n",
       "      <td>1075</td>\n",
       "      <td>1</td>\n",
       "      <td>2092.509298</td>\n",
       "      <td>5217.403538</td>\n",
       "      <td>2348.651952</td>\n",
       "      <td>4755.579746</td>\n",
       "      <td>1902.448739</td>\n",
       "      <td>689.823337</td>\n",
       "      <td>32714.431318</td>\n",
       "      <td>...</td>\n",
       "      <td>18275.561505</td>\n",
       "      <td>2764.023179</td>\n",
       "      <td>307.642235</td>\n",
       "      <td>52272.074896</td>\n",
       "      <td>1992.220640</td>\n",
       "      <td>1298.331547</td>\n",
       "      <td>1388.079434</td>\n",
       "      <td>35091.990798</td>\n",
       "      <td>611.218713</td>\n",
       "      <td>2337.809935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1062</td>\n",
       "      <td>1092</td>\n",
       "      <td>1</td>\n",
       "      <td>10993.829656</td>\n",
       "      <td>2478.139442</td>\n",
       "      <td>3871.937688</td>\n",
       "      <td>2526.369235</td>\n",
       "      <td>1892.057554</td>\n",
       "      <td>3776.799535</td>\n",
       "      <td>5219.417688</td>\n",
       "      <td>...</td>\n",
       "      <td>5777.510168</td>\n",
       "      <td>20434.521165</td>\n",
       "      <td>454.303864</td>\n",
       "      <td>13854.950602</td>\n",
       "      <td>3415.620723</td>\n",
       "      <td>1893.874800</td>\n",
       "      <td>680.118769</td>\n",
       "      <td>5727.192667</td>\n",
       "      <td>690.737793</td>\n",
       "      <td>5593.112143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>31504.607255</td>\n",
       "      <td>3255.911026</td>\n",
       "      <td>3730.017368</td>\n",
       "      <td>3048.706506</td>\n",
       "      <td>1048.029262</td>\n",
       "      <td>4071.288813</td>\n",
       "      <td>8457.360492</td>\n",
       "      <td>...</td>\n",
       "      <td>5938.515145</td>\n",
       "      <td>11401.686321</td>\n",
       "      <td>227.924972</td>\n",
       "      <td>25800.216417</td>\n",
       "      <td>2273.649598</td>\n",
       "      <td>1330.751430</td>\n",
       "      <td>434.642281</td>\n",
       "      <td>3956.475108</td>\n",
       "      <td>462.021279</td>\n",
       "      <td>5865.791959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>656</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21039.133213</td>\n",
       "      <td>2778.658293</td>\n",
       "      <td>2275.656845</td>\n",
       "      <td>3030.174777</td>\n",
       "      <td>1112.282795</td>\n",
       "      <td>1083.601090</td>\n",
       "      <td>24764.319265</td>\n",
       "      <td>...</td>\n",
       "      <td>7708.204866</td>\n",
       "      <td>2321.749289</td>\n",
       "      <td>617.759784</td>\n",
       "      <td>66079.910808</td>\n",
       "      <td>3860.368282</td>\n",
       "      <td>1629.795294</td>\n",
       "      <td>339.893955</td>\n",
       "      <td>20161.586529</td>\n",
       "      <td>563.680588</td>\n",
       "      <td>5327.787309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>22588.818351</td>\n",
       "      <td>3438.052885</td>\n",
       "      <td>3856.288323</td>\n",
       "      <td>2660.885013</td>\n",
       "      <td>1174.076044</td>\n",
       "      <td>7760.914622</td>\n",
       "      <td>10425.208266</td>\n",
       "      <td>...</td>\n",
       "      <td>4138.766957</td>\n",
       "      <td>2581.266569</td>\n",
       "      <td>191.600944</td>\n",
       "      <td>26637.841436</td>\n",
       "      <td>2515.928494</td>\n",
       "      <td>1228.068759</td>\n",
       "      <td>572.657187</td>\n",
       "      <td>5504.128198</td>\n",
       "      <td>409.904026</td>\n",
       "      <td>7891.519010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>18752.923297</td>\n",
       "      <td>3383.866368</td>\n",
       "      <td>2979.236304</td>\n",
       "      <td>3475.303494</td>\n",
       "      <td>1069.758958</td>\n",
       "      <td>6539.832626</td>\n",
       "      <td>14655.016100</td>\n",
       "      <td>...</td>\n",
       "      <td>3496.915905</td>\n",
       "      <td>1633.697432</td>\n",
       "      <td>236.905281</td>\n",
       "      <td>38168.987424</td>\n",
       "      <td>2594.182098</td>\n",
       "      <td>1488.887348</td>\n",
       "      <td>459.956452</td>\n",
       "      <td>6856.094249</td>\n",
       "      <td>478.327002</td>\n",
       "      <td>5191.827723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>659</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21653.214034</td>\n",
       "      <td>3285.588236</td>\n",
       "      <td>3140.755937</td>\n",
       "      <td>2982.100166</td>\n",
       "      <td>962.057182</td>\n",
       "      <td>7247.898317</td>\n",
       "      <td>9185.452866</td>\n",
       "      <td>...</td>\n",
       "      <td>4589.744961</td>\n",
       "      <td>39650.364469</td>\n",
       "      <td>338.675093</td>\n",
       "      <td>29649.402087</td>\n",
       "      <td>2561.112574</td>\n",
       "      <td>1495.228542</td>\n",
       "      <td>730.162136</td>\n",
       "      <td>11120.494057</td>\n",
       "      <td>382.049381</td>\n",
       "      <td>5795.120404</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows × 736 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    RNASeq   WGS  Type           FGR         CD99         ARF5      SLC25A5  \\\n",
       "0     1058  1015     1   5069.580040  1578.827545  2992.460441  3395.968681   \n",
       "1     1059   106     1  22322.926466  2977.917867  5103.243995  3036.767940   \n",
       "2     1060  1073     1  13309.274821  2625.227716  3452.118435  2699.675801   \n",
       "3     1061  1075     1   2092.509298  5217.403538  2348.651952  4755.579746   \n",
       "4     1062  1092     1  10993.829656  2478.139442  3871.937688  2526.369235   \n",
       "..     ...   ...   ...           ...          ...          ...          ...   \n",
       "67     655   NaN     0  31504.607255  3255.911026  3730.017368  3048.706506   \n",
       "68     656   NaN     0  21039.133213  2778.658293  2275.656845  3030.174777   \n",
       "69     657   NaN     0  22588.818351  3438.052885  3856.288323  2660.885013   \n",
       "70     658   NaN     0  18752.923297  3383.866368  2979.236304  3475.303494   \n",
       "71     659   NaN     0  21653.214034  3285.588236  3140.755937  2982.100166   \n",
       "\n",
       "         POLR2J         IL32         RPS20  ...      MTATP6P1     LINC02972  \\\n",
       "0   1611.595663  2398.030516  18619.603402  ...  19719.677898   4990.868852   \n",
       "1   1406.053242  4941.322966   8062.225679  ...  10154.439528   3324.101383   \n",
       "2   1713.541827  3821.111384  18238.310869  ...   9669.544485  27923.568621   \n",
       "3   1902.448739   689.823337  32714.431318  ...  18275.561505   2764.023179   \n",
       "4   1892.057554  3776.799535   5219.417688  ...   5777.510168  20434.521165   \n",
       "..          ...          ...           ...  ...           ...           ...   \n",
       "67  1048.029262  4071.288813   8457.360492  ...   5938.515145  11401.686321   \n",
       "68  1112.282795  1083.601090  24764.319265  ...   7708.204866   2321.749289   \n",
       "69  1174.076044  7760.914622  10425.208266  ...   4138.766957   2581.266569   \n",
       "70  1069.758958  6539.832626  14655.016100  ...   3496.915905   1633.697432   \n",
       "71   962.057182  7247.898317   9185.452866  ...   4589.744961  39650.364469   \n",
       "\n",
       "       CHCHD10         EEF1G         BRK1     C17orf49  LOC124903659  \\\n",
       "0   373.719041  57813.834711  3010.063149  1122.335152   1186.151740   \n",
       "1   252.586598  26712.912781  2523.261991  1524.612704    220.297177   \n",
       "2   405.456436  47541.807734  3075.430963  1672.903361    743.801781   \n",
       "3   307.642235  52272.074896  1992.220640  1298.331547   1388.079434   \n",
       "4   454.303864  13854.950602  3415.620723  1893.874800    680.118769   \n",
       "..         ...           ...          ...          ...           ...   \n",
       "67  227.924972  25800.216417  2273.649598  1330.751430    434.642281   \n",
       "68  617.759784  66079.910808  3860.368282  1629.795294    339.893955   \n",
       "69  191.600944  26637.841436  2515.928494  1228.068759    572.657187   \n",
       "70  236.905281  38168.987424  2594.182098  1488.887348    459.956452   \n",
       "71  338.675093  29649.402087  2561.112574  1495.228542    730.162136   \n",
       "\n",
       "           RPL17      NDUFA7        EGLN2  \n",
       "0   18162.339198  499.646110  3780.520158  \n",
       "1    3295.036434  474.430542  6131.083961  \n",
       "2   15782.782156  604.229611  5009.622187  \n",
       "3   35091.990798  611.218713  2337.809935  \n",
       "4    5727.192667  690.737793  5593.112143  \n",
       "..           ...         ...          ...  \n",
       "67   3956.475108  462.021279  5865.791959  \n",
       "68  20161.586529  563.680588  5327.787309  \n",
       "69   5504.128198  409.904026  7891.519010  \n",
       "70   6856.094249  478.327002  5191.827723  \n",
       "71  11120.494057  382.049381  5795.120404  \n",
       "\n",
       "[72 rows x 736 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counts_CIGTV = NCounts_CIGT.merge(RNASeq_VCF_IDs[['Gene Exp IDs', 'WGS Sample IDs']], left_on='ID', right_on='Gene Exp IDs', how='left')\n",
    "Counts_CIGTV.drop('Gene Exp IDs', axis=1, inplace=True)\n",
    "Counts_CIGTV.rename(columns={'WGS Sample IDs': 'SNP'}, inplace=True)\n",
    "cols = [Counts_CIGTV.columns[0], 'SNP'] + [col for col in Counts_CIGTV.columns if col not in ['ID', 'SNP']]\n",
    "Counts_CIGTV = Counts_CIGTV[cols]\n",
    "Counts_CIGTV['SNP'] = Counts_CIGTV['SNP'].str.split(\"-\").str[-1]\n",
    "Counts_CIGTV.rename(columns={'ID': 'RNASeq', 'SNP': 'WGS'}, inplace=True)\n",
    "Counts_CIGTV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "196f23ff-c67a-4a9f-9198-0284f51cb2fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from cyvcf2 import VCF\n",
    "\n",
    "def parse_csq_header(vcf):\n",
    "    for header in vcf.header_iter():\n",
    "        info = header.info(extra=True)\n",
    "        if info.get('ID') == 'CSQ':\n",
    "            description = info['Description']\n",
    "            format_description = description.split('Format: ')[-1].strip('\"')\n",
    "            return format_description.split('|')\n",
    "    return []\n",
    "\n",
    "def vcf_to_df(vcf_file):\n",
    "    vcf = VCF(vcf_file)\n",
    "    csq_header = parse_csq_header(vcf)\n",
    "    samples = vcf.samples\n",
    "    \n",
    "    field_indices = {field: csq_header.index(field) for field in ['Consequence', 'IMPACT', 'Existing_variation', 'SYMBOL', 'Feature_type', 'Feature', 'gnomADe_AF', 'CADD_PHRED', 'ClinVar_CLNSIG', 'ClinVar_CLNREVSTAT', 'ClinVar_CLNDN'] if field in csq_header}\n",
    "    \n",
    "    records = []\n",
    "\n",
    "    for variant in vcf:\n",
    "        record = {\n",
    "            'CHROM': variant.CHROM,\n",
    "            'POS': variant.POS,\n",
    "            'ID': variant.ID,\n",
    "            'REF': variant.REF,\n",
    "            'ALT': ','.join(variant.ALT),\n",
    "            'QUAL': variant.QUAL,\n",
    "            'FILTER': variant.FILTER,\n",
    "        }\n",
    "        \n",
    "        for field in field_indices.keys():\n",
    "            record[field] = []\n",
    "\n",
    "        csq_annotations = variant.INFO.get('CSQ', [])\n",
    "        for annotation in csq_annotations.split(','):\n",
    "            fields = annotation.split('|')\n",
    "            for field, index in field_indices.items():\n",
    "                value = fields[index] if index < len(fields) else ''\n",
    "                record[field].append(value)\n",
    "                \n",
    "        for key in field_indices.keys():\n",
    "            record[key] = ','.join(set(filter(None, record[key])))\n",
    "        \n",
    "        for sample in samples:\n",
    "            genotype_info_str = variant.format('GT')[vcf.samples.index(sample)]\n",
    "            # Check if genotype is present and not './.:.:.:.:.'\n",
    "            is_genotype_present = 1 if not all(gt == './.' for gt in genotype_info_str) else 0\n",
    "            record[f'{sample}_GT_Presence'] = is_genotype_present\n",
    "\n",
    "        records.append(record)\n",
    "    \n",
    "    return pd.DataFrame(records)\n",
    "\n",
    "vcf_path = '/PATH/CVD-SNPs_VEP_annotated.vcf'\n",
    "vcf = vcf_to_df(vcf_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "334104bd-1559-4057-a20b-0c38d483542b",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_df = vcf.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ac54f4f7-0b08-4864-8fc1-ff4016a933c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(583, 112)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_df['CADD_PHRED'] = pd.to_numeric(vcf_df['CADD_PHRED'], errors='coerce')\n",
    "vcf_df = vcf_df[vcf_df['CADD_PHRED'] >= 10]\n",
    "vcf_df['gnomADe_AF'] = pd.to_numeric(vcf_df['gnomADe_AF'], errors='coerce')\n",
    "vcf_df = vcf_df[vcf_df['gnomADe_AF'] <= 0.001]\n",
    "vcf_df = vcf_df[vcf_df['QUAL'] >= 50]\n",
    "vcf_df = vcf_df.drop('QUAL', axis = 1)\n",
    "vcf_df = vcf_df.drop('FILTER', axis = 1)\n",
    "vcf_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "84818370-d6c5-4a02-9186-c3c1c527b496",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'Consequence', 'IMPACT',\n",
      "       'Existing_variation', 'SYMBOL', 'Feature_type',\n",
      "       ...\n",
      "       '860', '890', '917', '921', '922', '941', '947', '975', '986', '995'],\n",
      "      dtype='object', length=112)\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "def extract_id(column_name):\n",
    "    match = re.search(r'DNA-BR2-(\\d+)_', column_name)\n",
    "    if match:\n",
    "        return match.group(1)  # Return the ID portion of the match\n",
    "    return column_name\n",
    "\n",
    "vcf_df.columns = [extract_id(col) for col in vcf_df.columns]\n",
    "\n",
    "print(vcf_df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6f8f305d-6c30-4a7d-bffe-492fb3851202",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_df['Position'] = vcf_df['CHROM'].astype(str) + ':' + vcf_df['POS'].astype(str)\n",
    "vcf_df['Allele'] = vcf_df['REF'].astype(str) + '>' + vcf_df['ALT'].astype(str)\n",
    "\n",
    "# Rename columns\n",
    "vcf_df.rename(columns={\n",
    "    'Existing_variation': 'SNV',\n",
    "    'SYMBOL': 'Gene',\n",
    "    'gnomADe_AF': 'Allele Frequency',\n",
    "    'CADD_PHRED': 'CADD Phred'\n",
    "}, inplace = True)\n",
    "\n",
    "new_column_order = ['Position', 'Allele'] + [col for col in vcf_df.columns if col not in ['Position', 'Allele', 'CHROM', 'POS', 'REF', 'ALT']]\n",
    "vcf_df = vcf_df[new_column_order]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d33fbede-cf8e-47fd-8fd8-3c77f00ff915",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_snv_names(snv_str):\n",
    "    snv_list = snv_str.split(',')\n",
    "    filtered_snv_list = [snv for snv in snv_list if snv.strip().startswith('rs')]\n",
    "    return ', '.join(filtered_snv_list)\n",
    "\n",
    "vcf_df['SNV'] = vcf_df['SNV'].apply(filter_snv_names)\n",
    "vcf_df.reset_index(drop=True, inplace=True)\n",
    "CIGT_df = vcf_df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1ed26588-f224-4158-baa6-c8128d861a93",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_df = vcf_df[['SNV', 'Position', 'Allele', 'Feature', 'Consequence', 'Allele Frequency', 'CADD Phred', 'Gene', 'ClinVar_CLNSIG', 'ClinVar_CLNREVSTAT', 'ClinVar_CLNDN']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "12f72438-ab58-43f1-8bb3-1b19738f597b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNV</th>\n",
       "      <th>Position</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149786471</td>\n",
       "      <td>chr13:27424869</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>ENST00000435781, ENST00000470606, ENST00000493...</td>\n",
       "      <td>Intron, NC Transcript, NMD Transcript, Synonym...</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>12.01</td>\n",
       "      <td>GTF3A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs201796232</td>\n",
       "      <td>chr13:27623037</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>ENST00000399696, ENST00000685267, ENST00000636...</td>\n",
       "      <td>Downstream Gene, Intron, NC Transcript, NMD Tr...</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>12.62</td>\n",
       "      <td>POLR1D,LNX2</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs41323349&amp;COSV100996298</td>\n",
       "      <td>chr13:30764017</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>ENST00000380490, ENST00000617770</td>\n",
       "      <td>Missense</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>18.94</td>\n",
       "      <td>ALOX5AP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1368294658&amp;COSV57290330</td>\n",
       "      <td>chr13:110642350</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>ENST00000257347, ENST00000537386, ENST00000535...</td>\n",
       "      <td>3'-UTR, Downstream Gene, Missense, NC Transcri...</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>29.20</td>\n",
       "      <td>CARS2,NAXD</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Combined_oxidative_phosphorylation_defect_type_27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs146355429</td>\n",
       "      <td>chr13:110642449</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>ENST00000257347, ENST00000537386, ENST00000535...</td>\n",
       "      <td>3'-UTR, Downstream Gene, Missense, NC Transcri...</td>\n",
       "      <td>0.000788</td>\n",
       "      <td>22.50</td>\n",
       "      <td>CARS2,NAXD</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Combined_oxidative_phosphorylation_defect_type...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>rs140891101</td>\n",
       "      <td>chr15:74926878</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>ENST00000564811, ENST00000322347, ENST00000562...</td>\n",
       "      <td>5'-UTR, Intron, Missense</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>24.70</td>\n",
       "      <td>COX5A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>rs762051874</td>\n",
       "      <td>chr15:82540457</td>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>ENST00000611163, ENST00000611031, ENST00000618...</td>\n",
       "      <td>5'-UTR, Downstream Gene, NC Transcript Exon, U...</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>12.11</td>\n",
       "      <td>CPEB1,RPS17</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>rs942454077</td>\n",
       "      <td>chr15:92974840</td>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>ENST00000625463, ENST00000636410, ENST00000700...</td>\n",
       "      <td>Downstream Gene, Intron, NC Transcript, NMD Tr...</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>15.79</td>\n",
       "      <td>CHD2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>rs202244266</td>\n",
       "      <td>chr15:92980915</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>ENST00000625463, ENST00000637789, ENST00000700...</td>\n",
       "      <td>Downstream Gene, Intron, NC Transcript, NMD Tr...</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>18.30</td>\n",
       "      <td>CHD2</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>not_provided&amp;Inborn_genetic_diseases&amp;Developme...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>rs755898320</td>\n",
       "      <td>chr15:93009265</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>ENST00000629104, ENST00000625662, ENST00000394...</td>\n",
       "      <td>3'-UTR, Downstream Gene, Missense, NC Transcri...</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>29.80</td>\n",
       "      <td>CHD2</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided&amp;_conflicting_classifications</td>\n",
       "      <td>Childhood_epilepsy_with_centrotemporal_spikes&amp;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>583 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           SNV         Position Allele  \\\n",
       "0                  rs149786471   chr13:27424869    C>T   \n",
       "1                  rs201796232   chr13:27623037    G>A   \n",
       "2     rs41323349&COSV100996298   chr13:30764017    T>C   \n",
       "3    rs1368294658&COSV57290330  chr13:110642350    G>A   \n",
       "4                  rs146355429  chr13:110642449    C>T   \n",
       "..                         ...              ...    ...   \n",
       "578                rs140891101   chr15:74926878    G>A   \n",
       "579                rs762051874   chr15:82540457    C>G   \n",
       "580                rs942454077   chr15:92974840    C>G   \n",
       "581                rs202244266   chr15:92980915    A>G   \n",
       "582                rs755898320   chr15:93009265    C>T   \n",
       "\n",
       "                                               Feature  \\\n",
       "0    ENST00000435781, ENST00000470606, ENST00000493...   \n",
       "1    ENST00000399696, ENST00000685267, ENST00000636...   \n",
       "2                     ENST00000380490, ENST00000617770   \n",
       "3    ENST00000257347, ENST00000537386, ENST00000535...   \n",
       "4    ENST00000257347, ENST00000537386, ENST00000535...   \n",
       "..                                                 ...   \n",
       "578  ENST00000564811, ENST00000322347, ENST00000562...   \n",
       "579  ENST00000611163, ENST00000611031, ENST00000618...   \n",
       "580  ENST00000625463, ENST00000636410, ENST00000700...   \n",
       "581  ENST00000625463, ENST00000637789, ENST00000700...   \n",
       "582  ENST00000629104, ENST00000625662, ENST00000394...   \n",
       "\n",
       "                                           Consequence  Allele Frequency  \\\n",
       "0    Intron, NC Transcript, NMD Transcript, Synonym...          0.000564   \n",
       "1    Downstream Gene, Intron, NC Transcript, NMD Tr...          0.000052   \n",
       "2                                             Missense          0.000143   \n",
       "3    3'-UTR, Downstream Gene, Missense, NC Transcri...          0.000006   \n",
       "4    3'-UTR, Downstream Gene, Missense, NC Transcri...          0.000788   \n",
       "..                                                 ...               ...   \n",
       "578                           5'-UTR, Intron, Missense          0.000076   \n",
       "579  5'-UTR, Downstream Gene, NC Transcript Exon, U...          0.000253   \n",
       "580  Downstream Gene, Intron, NC Transcript, NMD Tr...          0.000004   \n",
       "581  Downstream Gene, Intron, NC Transcript, NMD Tr...          0.000273   \n",
       "582  3'-UTR, Downstream Gene, Missense, NC Transcri...          0.000020   \n",
       "\n",
       "     CADD Phred         Gene                                ClinVar_CLNSIG  \\\n",
       "0         12.01        GTF3A                                                 \n",
       "1         12.62  POLR1D,LNX2                                 Likely_benign   \n",
       "2         18.94      ALOX5AP                                                 \n",
       "3         29.20   CARS2,NAXD                        Uncertain_significance   \n",
       "4         22.50   CARS2,NAXD                                        Benign   \n",
       "..          ...          ...                                           ...   \n",
       "578       24.70        COX5A                                                 \n",
       "579       12.11  CPEB1,RPS17                                                 \n",
       "580       15.79         CHD2                                                 \n",
       "581       18.30         CHD2                          Benign/Likely_benign   \n",
       "582       29.80         CHD2  Conflicting_classifications_of_pathogenicity   \n",
       "\n",
       "                                    ClinVar_CLNREVSTAT  \\\n",
       "0                                                        \n",
       "1                  criteria_provided&_single_submitter   \n",
       "2                                                        \n",
       "3                  criteria_provided&_single_submitter   \n",
       "4    criteria_provided&_multiple_submitters&_no_con...   \n",
       "..                                                 ...   \n",
       "578                                                      \n",
       "579                                                      \n",
       "580                                                      \n",
       "581  criteria_provided&_multiple_submitters&_no_con...   \n",
       "582     criteria_provided&_conflicting_classifications   \n",
       "\n",
       "                                         ClinVar_CLNDN  \n",
       "0                                                       \n",
       "1                                         not_provided  \n",
       "2                                                       \n",
       "3    Combined_oxidative_phosphorylation_defect_type_27  \n",
       "4    Combined_oxidative_phosphorylation_defect_type...  \n",
       "..                                                 ...  \n",
       "578                                                     \n",
       "579                                                     \n",
       "580                                                     \n",
       "581  not_provided&Inborn_genetic_diseases&Developme...  \n",
       "582  Childhood_epilepsy_with_centrotemporal_spikes&...  \n",
       "\n",
       "[583 rows x 11 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Category mapping\n",
    "category_mapping = {\n",
    "    '3_prime_UTR_variant': \"3'-UTR\",\n",
    "    '5_prime_UTR_variant': \"5'-UTR\",\n",
    "    'NMD_transcript_variant': \"NMD Transcript\",\n",
    "    'downstream_gene_variant': \"Downstream Gene\",\n",
    "    'intron_variant': \"Intron\",\n",
    "    'mature_miRNA_variant': \"Mature miRNA\",\n",
    "    'missense_variant': \"Missense\",\n",
    "    'non_coding_transcript_exon_variant': \"NC Transcript Exon\",\n",
    "    'non_coding_transcript_variant': \"NC Transcript\",\n",
    "    'splice_acceptor_variant': \"Splice Acceptor\",\n",
    "    'splice_donor_5th_base_variant': \"Splice Donor 5th Base\",\n",
    "    'splice_donor_region_variant': \"Splice Donor Region\",\n",
    "    'splice_polypyrimidine_tract_variant': \"Splice Polypyrimidine Tract\",\n",
    "    'splice_region_variant': \"Splice Region\",\n",
    "    'start_lost': \"Start Lost\",\n",
    "    'stop_gained': \"Stop Gained\",\n",
    "    'stop_retained_variant': \"Stop Retained\",\n",
    "    'synonymous_variant': \"Synonymous\",\n",
    "    'upstream_gene_variant': \"Upstream Gene\"\n",
    "}\n",
    "\n",
    "def clean_consequence(consequence):\n",
    "    unique_consequences = sorted(set(\n",
    "        category_mapping.get(part, part) \n",
    "        for part in consequence.replace('&', ',').split(',')\n",
    "    ))\n",
    "    return ', '.join(unique_consequences)\n",
    "\n",
    "vcf_df['Consequence'] = vcf_df['Consequence'].apply(clean_consequence)\n",
    "\n",
    "def add_space_after_commas(feature):\n",
    "    return '|'.join([part.replace(',', ', ') for part in feature.split('|')])\n",
    "\n",
    "vcf_df['Feature'] = vcf_df['Feature'].apply(add_space_after_commas)\n",
    "vcf_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b7306616-350c-418c-892e-423b7c642987",
   "metadata": {},
   "outputs": [],
   "source": [
    "NCounts_DE_CIGT = pd.read_csv('Results/NCounts_DE_CIGT.csv')\n",
    "RNASeq_VCF_IDs = pd.read_excel('Datasets/RNA-Seq_VCF_IDs.xlsx')\n",
    "RNASeq_VCF_IDs['Gene Exp IDs'] = RNASeq_VCF_IDs['Gene Exp IDs'].astype('Int64')\n",
    "RNASeq_VCF_IDs.drop('RNA-seq Sample IDs', axis = 1, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9f584257-2a8a-4b0c-a136-ec32a29d12dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "NCounts_DE_CIGT = NCounts_DE_CIGT.merge(RNASeq_VCF_IDs[['Gene Exp IDs', 'WGS Sample IDs']], left_on='ID', right_on='Gene Exp IDs', how='left')\n",
    "NCounts_DE_CIGT.drop('Gene Exp IDs', axis=1, inplace=True)\n",
    "NCounts_DE_CIGT.rename(columns={'WGS Sample IDs': 'SNP'}, inplace=True)\n",
    "cols = [NCounts_DE_CIGT.columns[0], 'SNP'] + [col for col in NCounts_DE_CIGT.columns if col not in ['ID', 'SNP']]\n",
    "NCounts_DE_CIGT = NCounts_DE_CIGT[cols]\n",
    "NCounts_DE_CIGT['SNP'] = NCounts_DE_CIGT['SNP'].str.split(\"-\").str[-1]\n",
    "NCounts_DE_CIGT.rename(columns={'ID': 'RNASeq', 'SNP': 'WGS'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8d6d950f-c5a5-45bd-b69d-b6953e9309c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "SNVs_CADD = pd.read_csv('/home/wbd20/Ahmed/Results/SNVs_CADD.csv')\n",
    "SNVs_CIGT = pd.read_csv('/home/wbd20/Ahmed/Results/SNVs_CIGT.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3a51fc33-8384-46e1-b884-2cdd7d692616",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WGS</th>\n",
       "      <th>rs750853525</th>\n",
       "      <th>rs149483341</th>\n",
       "      <th>rs764894987</th>\n",
       "      <th>rs141799330</th>\n",
       "      <th>rs61737078</th>\n",
       "      <th>rs115891972</th>\n",
       "      <th>rs759402008</th>\n",
       "      <th>rs773349568</th>\n",
       "      <th>rs776267089</th>\n",
       "      <th>...</th>\n",
       "      <th>rs1455280111</th>\n",
       "      <th>rs147644033</th>\n",
       "      <th>rs141524982</th>\n",
       "      <th>rs760078095</th>\n",
       "      <th>rs777136283</th>\n",
       "      <th>rs142659476</th>\n",
       "      <th>rs373979544</th>\n",
       "      <th>rs2229568</th>\n",
       "      <th>rs751011909</th>\n",
       "      <th>rs201003183</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1015</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1073</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1075</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>941</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>947</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>975</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>986</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>995</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     WGS  rs750853525  rs149483341  rs764894987  rs141799330  rs61737078  \\\n",
       "0   1007            0            0            0            0           0   \n",
       "1   1015            0            0            0            0           0   \n",
       "2    106            0            0            0            0           0   \n",
       "3   1073            0            0            0            0           0   \n",
       "4   1075            0            0            0            0           0   \n",
       "..   ...          ...          ...          ...          ...         ...   \n",
       "91   941            0            0            0            0           0   \n",
       "92   947            0            0            0            0           0   \n",
       "93   975            0            0            0            0           0   \n",
       "94   986            0            0            0            0           0   \n",
       "95   995            0            0            0            0           0   \n",
       "\n",
       "    rs115891972  rs759402008  rs773349568  rs776267089  ...  rs1455280111  \\\n",
       "0             0            0            0            0  ...             1   \n",
       "1             0            0            0            0  ...             1   \n",
       "2             0            0            0            0  ...             1   \n",
       "3             0            0            0            0  ...             1   \n",
       "4             0            0            0            0  ...             1   \n",
       "..          ...          ...          ...          ...  ...           ...   \n",
       "91            0            0            0            0  ...             1   \n",
       "92            0            0            0            0  ...             1   \n",
       "93            0            0            0            0  ...             1   \n",
       "94            0            0            0            0  ...             1   \n",
       "95            0            0            0            0  ...             1   \n",
       "\n",
       "    rs147644033  rs141524982  rs760078095  rs777136283  rs142659476  \\\n",
       "0             0            0            0            0            0   \n",
       "1             0            0            0            0            0   \n",
       "2             0            0            0            0            0   \n",
       "3             0            0            0            0            0   \n",
       "4             0            0            0            0            0   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "91            0            0            0            0            0   \n",
       "92            0            0            0            0            0   \n",
       "93            0            0            0            0            0   \n",
       "94            0            0            0            0            0   \n",
       "95            0            0            0            0            0   \n",
       "\n",
       "    rs373979544  rs2229568  rs751011909  rs201003183  \n",
       "0             0          0            0            0  \n",
       "1             0          0            0            0  \n",
       "2             0          0            0            0  \n",
       "3             0          0            0            0  \n",
       "4             0          0            0            0  \n",
       "..          ...        ...          ...          ...  \n",
       "91            0          0            0            0  \n",
       "92            0          0            0            0  \n",
       "93            0          0            0            0  \n",
       "94            0          0            0            0  \n",
       "95            0          0            0            0  \n",
       "\n",
       "[96 rows x 25 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes = NCounts_DE_CIGT.columns.drop(['RNASeq', 'WGS', 'Type']).unique()\n",
    "relevant_SNVs = SNVs_CADD[SNVs_CADD['Gene'].apply(lambda x: any(gene in x.split(', ') for gene in unique_genes))]\n",
    "relevant_SNV_ids = relevant_SNVs['SNV'].tolist()\n",
    "relevant_SNVs_CIGT = SNVs_CIGT[['WGS'] + [snv for snv in relevant_SNV_ids if snv in SNVs_CIGT.columns]]\n",
    "relevant_SNVs_CIGT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c10e50e3-8afb-4d8c-98c9-0de2a2881ac6",
   "metadata": {},
   "outputs": [],
   "source": [
    "NCounts_DE_CIGT = pd.read_csv('/home/wbd20/Ahmed/Results/NCounts_MRMR_CIGT.csv')\n",
    "RNASeq_VCF_IDs = pd.read_excel('Datasets/RNA-Seq_VCF_IDs.xlsx')\n",
    "RNASeq_VCF_IDs['Gene Exp IDs'] = RNASeq_VCF_IDs['Gene Exp IDs'].astype('Int64')\n",
    "RNASeq_VCF_IDs.drop('RNA-seq Sample IDs', axis = 1, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a5a8d677-8c2b-418c-a5eb-7d2590b71bf0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_92981/2749185975.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  relevant_SNVs_CIGT['WGS'] = pd.to_numeric(relevant_SNVs_CIGT['WGS'], errors='coerce', downcast = 'integer')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNASeq</th>\n",
       "      <th>WGS</th>\n",
       "      <th>Type</th>\n",
       "      <th>ITGB2</th>\n",
       "      <th>CD37</th>\n",
       "      <th>RPL36AP37</th>\n",
       "      <th>PSAP</th>\n",
       "      <th>ACTB</th>\n",
       "      <th>SELL</th>\n",
       "      <th>NCF2</th>\n",
       "      <th>...</th>\n",
       "      <th>rs776267089</th>\n",
       "      <th>rs576978943</th>\n",
       "      <th>rs199926951</th>\n",
       "      <th>rs147644033</th>\n",
       "      <th>rs141524982</th>\n",
       "      <th>rs760078095</th>\n",
       "      <th>rs142659476</th>\n",
       "      <th>rs373979544</th>\n",
       "      <th>rs751011909</th>\n",
       "      <th>rs201003183</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1058</td>\n",
       "      <td>1015.0</td>\n",
       "      <td>1</td>\n",
       "      <td>9134.766230</td>\n",
       "      <td>7351.580533</td>\n",
       "      <td>3050.136570</td>\n",
       "      <td>11776.206212</td>\n",
       "      <td>24872.277581</td>\n",
       "      <td>11468.465632</td>\n",
       "      <td>12347.235956</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1059</td>\n",
       "      <td>106.0</td>\n",
       "      <td>1</td>\n",
       "      <td>21651.847329</td>\n",
       "      <td>3313.983213</td>\n",
       "      <td>335.338536</td>\n",
       "      <td>48435.640379</td>\n",
       "      <td>105542.384031</td>\n",
       "      <td>29481.631763</td>\n",
       "      <td>19360.638770</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1060</td>\n",
       "      <td>1073.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12897.487026</td>\n",
       "      <td>4429.497525</td>\n",
       "      <td>1678.277112</td>\n",
       "      <td>22329.866174</td>\n",
       "      <td>4280.084624</td>\n",
       "      <td>27834.400439</td>\n",
       "      <td>12748.321271</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1061</td>\n",
       "      <td>1075.0</td>\n",
       "      <td>1</td>\n",
       "      <td>9729.447843</td>\n",
       "      <td>7967.124854</td>\n",
       "      <td>4070.316496</td>\n",
       "      <td>6120.981509</td>\n",
       "      <td>13905.797089</td>\n",
       "      <td>7112.157221</td>\n",
       "      <td>11393.281912</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1062</td>\n",
       "      <td>1092.0</td>\n",
       "      <td>1</td>\n",
       "      <td>22355.906429</td>\n",
       "      <td>3050.175942</td>\n",
       "      <td>1312.073643</td>\n",
       "      <td>37438.303304</td>\n",
       "      <td>7875.678779</td>\n",
       "      <td>45176.189087</td>\n",
       "      <td>4962.449512</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>20799.591680</td>\n",
       "      <td>5718.646289</td>\n",
       "      <td>170.273300</td>\n",
       "      <td>53181.103831</td>\n",
       "      <td>80117.340814</td>\n",
       "      <td>36387.340478</td>\n",
       "      <td>15872.934765</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>656</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>11585.467927</td>\n",
       "      <td>81613.963031</td>\n",
       "      <td>1439.657096</td>\n",
       "      <td>34188.621222</td>\n",
       "      <td>153016.433991</td>\n",
       "      <td>91570.387478</td>\n",
       "      <td>13791.901850</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21099.910461</td>\n",
       "      <td>5734.937774</td>\n",
       "      <td>138.279940</td>\n",
       "      <td>35868.948024</td>\n",
       "      <td>70836.550490</td>\n",
       "      <td>26929.563895</td>\n",
       "      <td>15827.843836</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>21967.843880</td>\n",
       "      <td>6035.958989</td>\n",
       "      <td>177.796498</td>\n",
       "      <td>44565.599462</td>\n",
       "      <td>88452.606286</td>\n",
       "      <td>25949.879057</td>\n",
       "      <td>15038.488443</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>659</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>27746.601258</td>\n",
       "      <td>6000.833127</td>\n",
       "      <td>205.816899</td>\n",
       "      <td>51271.408514</td>\n",
       "      <td>105743.167808</td>\n",
       "      <td>26109.397834</td>\n",
       "      <td>15126.516198</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    RNASeq     WGS  Type         ITGB2          CD37    RPL36AP37  \\\n",
       "0     1058  1015.0     1   9134.766230   7351.580533  3050.136570   \n",
       "1     1059   106.0     1  21651.847329   3313.983213   335.338536   \n",
       "2     1060  1073.0     1  12897.487026   4429.497525  1678.277112   \n",
       "3     1061  1075.0     1   9729.447843   7967.124854  4070.316496   \n",
       "4     1062  1092.0     1  22355.906429   3050.175942  1312.073643   \n",
       "..     ...     ...   ...           ...           ...          ...   \n",
       "67     655     NaN     0  20799.591680   5718.646289   170.273300   \n",
       "68     656     NaN     0  11585.467927  81613.963031  1439.657096   \n",
       "69     657     NaN     0  21099.910461   5734.937774   138.279940   \n",
       "70     658     NaN     0  21967.843880   6035.958989   177.796498   \n",
       "71     659     NaN     0  27746.601258   6000.833127   205.816899   \n",
       "\n",
       "            PSAP           ACTB          SELL          NCF2  ...  rs776267089  \\\n",
       "0   11776.206212   24872.277581  11468.465632  12347.235956  ...          0.0   \n",
       "1   48435.640379  105542.384031  29481.631763  19360.638770  ...          0.0   \n",
       "2   22329.866174    4280.084624  27834.400439  12748.321271  ...          0.0   \n",
       "3    6120.981509   13905.797089   7112.157221  11393.281912  ...          0.0   \n",
       "4   37438.303304    7875.678779  45176.189087   4962.449512  ...          0.0   \n",
       "..           ...            ...           ...           ...  ...          ...   \n",
       "67  53181.103831   80117.340814  36387.340478  15872.934765  ...          NaN   \n",
       "68  34188.621222  153016.433991  91570.387478  13791.901850  ...          NaN   \n",
       "69  35868.948024   70836.550490  26929.563895  15827.843836  ...          NaN   \n",
       "70  44565.599462   88452.606286  25949.879057  15038.488443  ...          NaN   \n",
       "71  51271.408514  105743.167808  26109.397834  15126.516198  ...          NaN   \n",
       "\n",
       "    rs576978943  rs199926951  rs147644033  rs141524982  rs760078095  \\\n",
       "0           0.0          0.0          0.0          0.0          0.0   \n",
       "1           0.0          0.0          0.0          0.0          0.0   \n",
       "2           0.0          0.0          0.0          0.0          0.0   \n",
       "3           0.0          0.0          0.0          0.0          0.0   \n",
       "4           0.0          0.0          0.0          0.0          0.0   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "67          NaN          NaN          NaN          NaN          NaN   \n",
       "68          NaN          NaN          NaN          NaN          NaN   \n",
       "69          NaN          NaN          NaN          NaN          NaN   \n",
       "70          NaN          NaN          NaN          NaN          NaN   \n",
       "71          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    rs142659476  rs373979544  rs751011909  rs201003183  \n",
       "0           0.0          0.0          0.0          0.0  \n",
       "1           0.0          0.0          0.0          0.0  \n",
       "2           0.0          0.0          0.0          0.0  \n",
       "3           0.0          0.0          0.0          0.0  \n",
       "4           0.0          1.0          0.0          0.0  \n",
       "..          ...          ...          ...          ...  \n",
       "67          NaN          NaN          NaN          NaN  \n",
       "68          NaN          NaN          NaN          NaN  \n",
       "69          NaN          NaN          NaN          NaN  \n",
       "70          NaN          NaN          NaN          NaN  \n",
       "71          NaN          NaN          NaN          NaN  \n",
       "\n",
       "[72 rows x 30 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NCounts_DE_CIGT = NCounts_DE_CIGT.merge(RNASeq_VCF_IDs[['Gene Exp IDs', 'WGS Sample IDs']], left_on='ID', right_on='Gene Exp IDs', how='left')\n",
    "NCounts_DE_CIGT.drop('Gene Exp IDs', axis=1, inplace=True)\n",
    "NCounts_DE_CIGT.rename(columns={'WGS Sample IDs': 'SNP'}, inplace=True)\n",
    "cols = [NCounts_DE_CIGT.columns[0], 'SNP'] + [col for col in NCounts_DE_CIGT.columns if col not in ['ID', 'SNP']]\n",
    "NCounts_DE_CIGT = NCounts_DE_CIGT[cols]\n",
    "NCounts_DE_CIGT['SNP'] = NCounts_DE_CIGT['SNP'].str.split(\"-\").str[-1]\n",
    "NCounts_DE_CIGT.rename(columns={'ID': 'RNASeq', 'SNP': 'WGS'}, inplace=True)\n",
    "NCounts_DE_CIGT['WGS'] = pd.to_numeric(NCounts_DE_CIGT['WGS'], errors='coerce', downcast = 'integer')\n",
    "relevant_SNVs_CIGT['WGS'] = pd.to_numeric(relevant_SNVs_CIGT['WGS'], errors='coerce', downcast = 'integer')\n",
    "merged_df = pd.merge(NCounts_DE_CIGT, relevant_SNVs_CIGT, on = \"WGS\", how = \"left\")\n",
    "\n",
    "df = merged_df\n",
    "columns_to_check = [col for col in df.columns if col.startswith('rs')]\n",
    "for col in columns_to_check:\n",
    "    if df[col].sum() == 0 or df[col].sum() >= 2:\n",
    "        df.drop(col, axis=1, inplace=True)\n",
    "\n",
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
